dify/weaviate_docs.py

```diff
@@ -1,103 +1,348 @@
+        splitter = RecursiveCharacterTextSplitter(
+            chunk_size=chunk_size, chunk_overlap=chunk_overlap
+        )
+
+        # Track processing results.
+        successful_files = 0
+        failed_files = 0
+        total_files = 0
+
+        # Process files in batches.
+        _LOG.info("Starting batch processing with batch size: %s", batch_size)
+        with collection.batch.fixed_size(batch_size) as batch:
+            for root, _, files in os.walk(docs_dir):
+                for filename in files:
+                    # Filter by allowed extensions.
+                    if not any(
+                        filename.endswith(ext) for ext in allowed_extensions
+                    ):
+                        continue
+
+                    total_files += 1
+                    filepath = os.path.join(root, filename)
+
+                    # Process the file.
+                    if process_markdown_file(
+                        filepath,
+                        docs_dir,
+                        splitter,
+                        batch,
+                        ollama_url,
+                        ollama_model,
                     ):
-                        raise ValueError(
-                            "Embedding format unexpected or missing."
-                        )
-                    embedding = embedding_list[0]
-                    batch.add_object(
-                        properties={
-                            "text": text,
-                            "filename": filename,
-                            "filepath": os.path.relpath(filepath, DOCS_DIR),
-                        },
-                        vector=embedding,
-                    )
-                print(f"Uploaded: {filename}")
-            except Exception as e:
-                print(f"Failed: {filename}, error: {e}")
-
-client.close()
-print("All markdown files uploaded using `nomic-embed-text` via Ollama.")
+                        successful_files += 1
+                    else:
+                        failed_files += 1
+
+        # Summary.
+        result = {
+            "total_files": total_files,
+            "successful_files": successful_files,
+            "failed_files": failed_files,
+            "collection_name": collection_name,
+            "docs_directory": docs_dir,
+        }
+
+        _LOG.info("Upload process completed")
+        _LOG.info("Total files processed: %s", total_files)
+        _LOG.info("Successful uploads: %s", successful_files)
+        _LOG.info("Failed uploads: %s", failed_files)
+        _LOG.info("Collection: %s", collection_name)
+        _LOG.info(
+            "All markdown files uploaded using '%s' via Ollama", ollama_model
+        )
+
+        return result
+
+    finally:
+        # Close client only if we created it.
+        if should_close_client:
+            client.close()
+            _LOG.debug("Weaviate client connection closed")
+
+
+def configure_logging(
```

Comment:
You can use ./helpers_root/dev_scripts_helpers/coding_tools/script_template.py as template